CONTRIBUTING.md

```diff
@@ -54,19 +54,198 @@ development.
+  deploy a JupyterHub using the instructions (make sure that
+  you're reading from `/en/latest`). Make sure your latest
+  changes are present, and that the JupyterHub successfully deploys
+  and functions properly.
+
+Next, move on to making a GitHub release, described below.
+
+### Tagging and making a GitHub release
+
+Now that our Docker image is pushed and we have updated the documentation
+for z2jh, it's time to make a new GitHub release. To do this, you must have:
+
+1. Push rights to the `jupyterhub/zero-to-jupyterhub-k8s` repo
+
+You will need to make a git tag, and then create a GitHub release.
+
+1. Make sure you're on branch `master` with your latest changes from
+  the section above pulled.
+2. Make a git tag with:
+  ```
+  git tag -a <VERSION>
+  ```
+
+  Where `<VERSION>` should be the new version that you're releasing.
+  Note the `v` before the version number.
+
+  Git will ask you to include a message with the tag.
+  Paste the entire contents of the CHANGELOG for this particular release.
+  An easy way to do this is to paste the contents in a text file, and
+  then refer to that text file with the call to commit:
+  `git tag -a <VERSION> -F <PATH-TO-FILE.txt>`
+3. Push the tags to the `jupyterhub/zero-to-jupyterhub-k8s` repo with
+  `git push <REMOTE-NAME> --tags`.
+  Note that `<REMOTE-NAME>` is whatever your local git uses to refer
+  to the `jupyerhub/` organization's repository (e.g., `official`
+  or `upstream`)
+3. Make a **GitHub Release**:
+  * go to https://github.com/jupyterhub/zero-to-jupyterhub-k8s/releases and click 'Draft new release'.
+  * The title should be the new version, followed by the name of the cricketer for the release. Like so:`v0.5: "Hamid Hassan"`.
+  * The description should include the entire changelog entry for this release.
+  * Make sure the title/description/tag name look correct, and then click
+    on `Publish Release`.
+
+You've just made a GitHub release!
+
+
+### RTD update
+
+Wait a few hours to let the release 'cool' and make sure that links,
+webpages, etc have updated. Then, update our documentation settings on
+readthedocs to show `latest` by default. This marks the official
+'release' of the version!
+
+### Last step - release a blog post and tell the world!
+
+The final step is to release a blog post. This doesn't have to be
+done by the person who performed all of the above actions.
+
+To release a blog post for the new version, start a draft on the Jupyter Medium
+blog. Copy/paste the section of the CHANGELOG corresponding to the new
+release, then make minor modifications to make it more blog-friendly.
+
+Don't forget to tell the JupyterHub community about the new release, and
+to encourage people to talk about it on social media!
+
+That's it! Congratulations on making a new release of JupyterHub!
+
+### Extra step - release a documentation release
+
+It is common that documentation changes are made shortly after a new release.
```

Comment:
I'm inclined to say let's just do this as part of the release process (making the time window: 2 weeks or when the next release docs begin to be written).

Otherwise, we are generating more work for us and more confusion for the end user (a large portion will looks for the current version docs).

Reply:
IIRC the reasoning was that there are still a lot of large-ish changes to the docs happening, so it'd be best if the docs always show the latest updates until there's something written that doesn't apply to the currently-released helm chart. 

What if we kept this behavior for one or two release cycles, and then once the docs stabilize we can start to just jump straight from `last-version` to `next-version`?

cc @yuvipanda in case you've got thoughts as well

Reply:
As long as the comment (Code, tests, and documentation to support a release are stable.) goes in as the first checklist step, I'm fine doing this as you propose for a couple of releases.